### DISCUSSION TITLE
Multiple onBeforeRender hooks

### DISCUSSION THREAD

**Discussion by mschakulat** [vikejs/vike#862] [Discussion] [answered]
Hi. I have a question about Support for multiple onBeforeRender() hooks is deprecated. I know that I can use custom exports, but I need the pageContext in the various *.page.server.js files. How can I achieve using the pageContext on different pages, i.e., different page.server.js files, without multiple onBeforeRender() hooks?

**Comment by brillout** [comment]
You can define one `onBeforeRender()` hook per page.

**Reply by mschakulat** [reply]
What about this deprecation warning?:
https://vite-plugin-ssr.com/onBeforeRender-multiple

**Reply by brillout** [reply, marked as answer]
https://github.com/brillout/vite-plugin-ssr/commit/85f51860e241afbce1a02585d06eecd258c92518. I hope it makes more sense.

Btw. https://github.com/sponsors/brillout &mdash; in case you think you can convince your employer.

**Reply by mschakulat** [reply]
That definitely makes sense. I'll solve the problem in a different way then. Thank you for your help.
